### PR TITLE
new functions

### PR DESCRIPTION
**Pull request by sergey-shandar**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -19,3 +25,15 @@ export const recursiveReaddir = (dir: string): asyncIt.AsyncIterableEx<string> =
         }
     )
 
+export const recursiveRmdir = async (dir: string): Promise<void> => {
```
Comment (vladbarosan) [inline src/index.ts:28]: ther eis already a de facto standard library for handlings FS stuff, I would not reinvent the wheel and use whats proven in the field https://github.com/jprichardson/node-fs-extra/blob/master/docs/remove.md

File: src/index.ts [hunk 2]
```diff
@@ -11,6 +11,12 @@ export const exists = util.promisify(fs.exists)
 
 export const readdir = util.promisify(fs.readdir)
 
+export const mkdir = util.promisify(fs.mkdir)
```
Comment (vladbarosan) [inline src/index.ts:14]: same for these ones, "fs-extra" has everything from here"

Comment (vladbarosan) [reply src/index.ts:14]: async and sync and with graceful handling of different type of errors etc

File: src/index.ts (hunk 1 above)
Comment (sergey-shandar) [reply src/index.ts:28]: TypeScript definitions are two versions behind 
https://www.npmjs.com/package/@types/fs-extra 7.0
https://www.npmjs.com/package/fs-extra 5.0

Comment (sergey-shandar) [reply src/index.ts:28]: And they don't have recursive readdir with async iterator. BTW, it has the same problem with TypeScript definitions that other JS libraries do. To write correct TypeScript definitions is really hard and requires much more work and testing than to write a new library for such small functions.

**Pull request merged**